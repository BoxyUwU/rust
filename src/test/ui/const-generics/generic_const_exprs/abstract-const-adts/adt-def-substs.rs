#![feature(generic_const_exprs, adt_const_params)]
#![allow(incomplete_features)]

#[derive(PartialEq, Eq)]
enum Foo<T: Eq + PartialEq> {
    Unused(T),
    Variant(u32),
}

struct HasFooU16<const N: Foo<u16>>;

fn doesnt_unify<const N: u32>(_: HasFooU16<{ Foo::<u16>::Variant(N) }>) {
    bar::<{ Foo::<u8>::Variant(N) }>();
    //~^ error: unconstrained generic constant
}

struct HasFooU8<const N: Foo<u8>>;

fn does_unify<const N: u32>(_: HasFooU8<{ Foo::<u8>::Variant(N) }>) {
    bar::<{ Foo::<u8>::Variant(N) }>();
}

fn bar<const M: Foo<u8>>() {}

fn main() {}
