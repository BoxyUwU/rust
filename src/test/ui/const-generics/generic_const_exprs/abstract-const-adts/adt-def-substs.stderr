error: unconstrained generic constant
  --> $DIR/adt-def-substs.rs:13:11
   |
LL |     bar::<{ Foo::<u8>::Variant(N) }>();
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: try adding a `where` bound using this expression: `where [(); { Foo::<u8>::Variant(N) }]:`

error: aborting due to previous error

